### DISCUSSION TITLE
Failed to identify JPM file

### DISCUSSION THREAD

**Discussion by LyAnhVu** [ImageMagick/ImageMagick#7862] [Development]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows

### Description

I use ImageMagick to identify and convert JPM file but get the error messages:
ERROR: [JP2H box missing. Required.
 'OpenJP2' @ error/jp2.c/JP2ErrorHandler/197] [(null)]
ERROR: [UnableToDecodeImageFile 'balloon.jpm' @ error/jp2.c/ReadJP2Image/363] [(null)]
ERROR: [NoImagesDefined 'jp2:balloon_out.jpm' @ error/convert.c/ConvertImageCommand/3366] [(null)]

I use command:
`magick.exe identify <input file>`
`magick.exe convert jpm:<input file> jpm:<output file>`

Please help me investigate it and let me know if any information/the reason why error appears.
Thanks a lot.
[jpm_sample.zip](https://github.com/user-attachments/files/18257113/jpm_sample.zip)

**Comment by dlemstra** [comment]
It looks like your file is written incorrectly and that is why it fails.

**Reply by LyAnhVu** [reply]
I think my file is valid structure because it can be opened with online tool: https://products.groupdocs.app/viewer/total and it also complies with JPM file format.

**Reply by fmw42** [reply]
IM is very strict to follow correct file specifications. Other tools are not so strict and can open your file.  Your file likely deviates from proper structure and IM won't open it.

**Reply by LyAnhVu** [reply]
Can you tell me where my file structure is wrong and provide me with a sample JPM file?? I tried with another JPM file collected from the Internet and got same error. 
[sample2.zip](https://github.com/user-attachments/files/18268615/sample2.zip)

**Reply by dlemstra** [reply]
This is an error message from the openjpeg decoder. It might be a bug on their side? This check was added recently and prevents us from reading that file. It might also be possible that reading JPM files worked by accident. I would advise you to contact them instead.

**Reply by LyAnhVu** [reply]
Thanks @dlemstra, I will contact them.
